gdal/swig/python/scripts/gdal2tiles.py

```diff
@@ -969,6 +969,13 @@ def create_base_tile(tile_job_info, tile_detail, queue=None):
                              band_list=list(range(1, dataBandsCount + 1)))
         alpha = alphaband.ReadRaster(rx, ry, rxsize, rysize, wxsize, wysize)
 
+        # Detect totally transparent tile and skip its creation
+        if tile_job_info.exclude_transparent and len(alpha) == alpha.count('\x00'):
```

Comment:
I don't believe this will Python3 compatible. Using '\x00'.encode('ascii') should work for both Python 2 and 3

Reply:
a = '\x00\x00'
    print(len(a), a.count('\x00'))

    > 2 2

Reply:
the test is not relevant. ReadRaster() returns a binary buffer in Python 3, not a string

```
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from osgeo import gdal
>>> ds = gdal.Open('byte.tif')
>>> ds.ReadRaster().count('\x00')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
>>> ds.ReadRaster().count('\x00'.encode('ascii'))
0
>>> 
```

Reply:
You are right. Change updated.

Reply:
About test... I'm not sure I'm capable to do it right. I'll try, but if anybody wants to help with it I won't refuse =)